projects/packages/search/src/class-cli.php

```diff
@@ -32,4 +34,23 @@ public function auto_config() {
 			WP_CLI::error( $e->getMessage() );
 		}
 	}
+
+	/**
+	 * Choose a admin user and set it.
+	 */
+	protected function set_admin_user() {
+		$user = get_user_by( 'ID', 1 );
```

Comment:
I feel uncomfortable about doing this too tbh. There's somewhere in WP to create post to store the altered template part. When doing that, the current user is used to be the author of the post - if we don't set a current user, the post would be invisible due to some relationships not set ( template part belongs to which theme etc ) . 
Since it's CLI, I don't think there's a way to get the current user. Luckily the code could only be run from WPCLI, meaning it should only be run by a user privileged enough. 

The reason I prefer to use ID 1 is that if anything magic needs to be done, I would rather impersonate an initial admin.

Any suggestions appreciated, thanks.

Reply:
Absolutely good question btw.

Reply:
Now the CLI would require a user ID or login to run 3e87dff.